lib/cudadrv/stream.jl

```diff
@@ -153,13 +153,31 @@ end
         spins += 1
     end
 
-    # minimize CPU usage of long-running kernels
-    # by waiting for an event signalled by CUDA
-    event = Threads.Event()
+    # minimize CPU usage of long-running kernels by waiting for an event signalled by CUDA
+    event = Base.Event()
     launch(; stream) do
         notify(event)
     end
-    Base.wait(event)
+    # if an error occurs, the callback may never fire, so use a timer to detect such cases
+    dev = device()
+    timer = Timer(0; interval=1)
+    Base.@sync begin
+        @async begin
```

Comment:
Any reason to use `@async` instead of `@spawn`? `@async` will also pin the parent.

Reply:
Ah, didn't know that. I'm just used to old-style concurrent programming, but I guess we should be sufficiently threadsafe by now?